src/Sulu/Bundle/MarkupBundle/Markup/HtmlTagExtractor.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\MarkupBundle\Markup;
+
+/**
+ * Extracts tags from html content.
+ */
+class HtmlTagExtractor implements TagExtractorInterface
+{
+    const COUNT_REGEX = '/<%1$s:[a-z]+/';
+    const ATTRIBUTE_REGEX = '/(?<name>\b[\w-]+\b)\s*=\s*"(?<value>[^"]*)"/';
+    const CONTENT_REGEX = '/(?:>(?<content>[^<]*)<)/';
+    const TAG_REGEX = '/(?<tag><%1$s:(?<name>[a-z]+)((?!>|\/>).)*(?:\/>|>((?!<\/%1$s:\2*>).)*<\/%1$s:\2*>))/';
```

Comment:
I dunno if I'm wrong, but I think the 3rd and 4th Group doesn't have to be capturing?
`/(?<tag><%1$s:(?<name>[a-z]+)(?:(?!>|\/>).)*(?:\/>|>(?:(?!<\/%s:\2*>).)*<\/%1$s:\2*>))/`